client/src/components/FormField/RegisteredMFAMethodListField/RegisteredMFAMethodListField.js

```diff
@@ -1,12 +1,60 @@
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-
-import methodShape from '../../../types/registeredMethod';
+import { Button, Modal, ModalHeader, ModalBody } from 'reactstrap';
+import { inject } from 'lib/Injector'; // eslint-disable-line
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import methodShape from 'types/registeredMethod';
 import MethodListItem from './MethodListItem';
+import { showScreen, chooseMethod, setAvailableMethods } from 'state/mfaRegister/actions';
+import { SCREEN_CHOOSE_METHOD } from 'components/Register';
 
 const fallbacks = require('../../../../lang/src/en.json');
 
 class RegisteredMFAMethodListField extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      modalOpen: false,
+    };
+
+    props.onUpdateAvailableMethods(props.availableMethods);
```

Comment:
This is throwing React errors at me, and prevents me from going through the TOTP registration process successfully:

>Warning: setState(...): Cannot update during an existing state transition (such as within `render` or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`.

Reply:
We can easily replace this with componentDidMount. I still get this error when trying to go to the second screen of TOTP registration though: https://reactjs.org/warnings/refs-must-have-owner.html

In our case we are both creating a ref for a component that didn't exist in the initial render() AND using two versions of React. Not sure what the solution is here aside from trying to rewrite so it doesn't use a reference. From memory it's only to auto-focus the input field when the TOTP verification screen is rendered.

Reply:
Hmmmn I pulled the latest master of `totp` and I'm not getting this issue 🤔

Reply:
Maybe it was just me. Fine for now, can address later.